pkg/cli/debug/debug.go

```diff
@@ -1349,3 +1350,54 @@ func (o *DebugOptions) resolveImageStreamTag(namespace, name, tag string) (strin
 	}
 	return image, nil
 }
+
+// debugCompletionFunc Returns a completion function that completes:
+// 1- pod names that match the toComplete prefix
+// 2- resource types containing pods and nodes which match the toComplete prefix
+func debugCompletionFunc(f kcmdutil.Factory) func(*cobra.Command, []string, string) ([]string, cobra.ShellCompDirective) {
+	return func(_ *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		var comps []string
+		directive := cobra.ShellCompDirectiveNoFileComp
+		if len(args) != 0 {
+			return comps, directive
+		}
+
+		directive = cobra.ShellCompDirectiveNoFileComp
+		slashIdx := strings.Index(toComplete, "/")
+		if slashIdx == -1 {
+			// Standard case, complete pod names
+			comps = completion.CompGetResource(f, "pod", toComplete)
+
+			validResources := []string{
```

Comment:
This list is not fully complete. oc debug supports resources https://github.com/openshift/oc/blob/beeaf0904b6e9a7c22a73af9e92327f50a4d6918/pkg/cli/debug/debug.go#L1021 more than this list
/hold

Reply:
Added imagestreamimages, imagestreamtags and cronjobs.
I didn't add deploymentConfig since it is deprecated since 4.14.

I can add it if you think it should be here.